src/Compiler/Service/FSharpCheckerResults.fs

```diff
@@ -2357,12 +2357,25 @@ module internal ParseAndCheckFile =
         let tokenizer =
             LexFilter.LexFilter(indentationSyntaxStatus, options.CompilingFSharpCore, Lexer.token lexargs true, lexbuf, false)
 
-        (fun _ -> tokenizer.GetToken())
+        if ct.CanBeCanceled then
```

Comment:
Is the check necessary? All it does, checks CTS for null, right? And the method already does that. In case if we will be calling that lambda with the check in it, JIT will likely optimize the call to be a nop if CTS is default/non-cancellable.

Reply:
Though it will ikely be inlined either way.

Reply:
The difference is that the code added by me is invoked just once, and later on the lambda is created.
If we are e.g. passed a CancellationToken.None , then just the plain lambda is returned and no subsequent checking happens at all, never.

TLDR:
The difference is between making the check just once before creating the func, or inside the func each time.

Reply:
I tested it - you are right under the circumstance of everything being inline from the place where token is being created.
(which might be well inside the VisualStudio / Rider /.. dlls).
In the inlined case, I can indeed see the "nop"s.

But when it is not inlined, the asm contains the full check.

https://sharplab.io/#v2:DYLgZgzgNALiCGEC2AfA9gBwKYDsAEAygJ4QxZIDcAsAFCa6EllIB0AKgBYBOW8AJgEscAc1piawLDDwAKAFRDgQrHICUeGBwEQAkhACyRAGIBXHAGNZ5uAGF4FrMGDwYAtDjZoA1rnUBeWjwgvElpJBMYeAAjSTwAdw4XAFEANywuPD88AEZA4LA0DIE/AAYNNBySqrw+NDzg4Ot2bjQ4nTA7BycXNxwAJSwARxMsUiw+GVV6hvjEmFT0vAAeAFpZ5LSMuTwAJmmG/eDaULwcVqNCgfhgWX9pk4APTI0tXQNjM0sZTvNHZ1d3J4fDgWAA5dxYKY0BoPaZwiRSU7nS68YA7W6Ze6IiBoExcX7PHBYOJ4H5/HqA7y4Ai4/FYSZY6RPLKabR6QymCyyHF437sKk4KEwoA=

=> Since the callstack up to this point is large and not controlled by us, I would rather not assume inlining happens and leave the code as is